networks/p2p/dial.go

```diff
@@ -132,6 +132,7 @@ type dialTask struct {
 	resolveDelay time.Duration
 	failedTry    int
 	dialType     dialType
+	err          error
```

Comment:
Is this error used only for `errUpdateDial`? It seems confusing since other errors are not handled. How about adding a boolean instead of error like `expirable`?

Reply:
Review reflected :) Thnx